src/plots/plots.js

```diff
@@ -715,6 +750,42 @@ plots.supplyDataDefaults = function(traceIn, traceIndex, layout) {
     return traceOut;
 };
 
+function supplyTransformDefaults(traceIn, traceOut, layout) {
+    var containerIn = traceIn.transforms || [],
+        containerOut = traceOut.transforms = [];
```

Comment:
I think this line here ensures that *every* fullTrace will now have a transforms array? Most of the time it will be empty. Some tests that have nothing to do with transforms are failing due to an empty transforms array appearing. I think this is the culprit.